templates/documents/file_upload_modal.html

```diff
@@ -76,26 +82,57 @@ <h3 class="font-bold text-lg mb-4">Add Files to Collection</h3>
                     </div>
                 </template>
             </div>
-            <div class="modal-action">
-                <button class="btn btn-primary mt-2" :disabled="uploaded_files.length === 0">Submit</button>
+            <div class="modal-action flex flex-row gap-2">
+                <span x-cloak x-show="filesTooMany > 0" class="text-error">
+                    Uploading <span x-text="uploaded_files.length"></span> files exceeds the limit of {{ max_files_per_collection }} files per collection. Please remove <span x-text="filesTooMany"></span> files
+                </span>
+                <button class="btn btn-primary mt-2" :disabled="!canSubmit()">Submit</button>
             </div>
         </form>
     </div>
 </dialog>
 
+{{ max_files_per_collection|json_script:"maxFilesPerCollection" }}
+{{ collection_files.count|json_script:"currentCollectionFileCount" }}
+{{ max_file_size_mb|json_script:"maxFileSizeMb" }}
 <script>
+    function convertMbToBytesBase10(valueMb) {
+        return 1000000 * valueMb; // Convert MB to bytes, using base 10
+    }
+
     document.addEventListener('alpine:init', () => {
         Alpine.data('modalData', () => ({
-            message: "",
+            maxFilesPerCollection: JSON.parse(document.getElementById("maxFilesPerCollection").textContent),
+            currentCollectionFileCount: JSON.parse(document.getElementById("currentCollectionFileCount").textContent),
+            maxFileSizeMB: JSON.parse(document.getElementById("maxFileSizeMb").textContent),
```

Comment:
since these are just integer values you could use them directly:
```suggestion
            maxFilesPerCollection: {{ max_files_per_collection }},
            currentCollectionFileCount: {{ collection_files.count }},
            maxFileSizeMB: {{ max_file_size_mb }},
```

Also it seems you're doing the count query twice, once in the view and once in the template. Maybe rather just assign another var.